src/Datadog.NativeLoader/pal.cpp

```diff
@@ -0,0 +1,43 @@
+#include "pal.h"
+
+#if _WIN32
+#include <Windows.h>
+#else
+#include "dlfcn.h"
+#endif
+
+#include "logging.h"
+
+namespace datadog
+{
+namespace nativeloader
+{
+
+#if _WIN32
+    void* LoadDynamicLibrary(std::string filePath)
+    {
+        Debug("LoadLibrary: ", filePath);
+        return LoadLibrary(ToWSTRING(filePath).c_str());
+    }
+
+    void* GetExternalFunction(void* instance, std::string funcName)
+    {
+        Debug("GetExternalFunction: ", funcName);
+        return (void*)GetProcAddress(instance, funcName.c_str());
+    }
+#else
+    void* LoadDynamicLibrary(std::string filePath)
+    {
+        Debug("LoadLibrary: ", filePath);
+        return dlopen(filePath.c_str(), RTLD_LOCAL | RTLD_LAZY);
+    }
+
+    void* GetExternalFunction(void* instance, std::string funcName)
+    {
+        Debug("GetExternalFunction: ", funcName);
+        return dlsym(instance, funcName.c_str());
```

Comment:
returning null is not an error per se. According to the documentation:
```
, the correct way to test for an error is to call dlerror() to clear any old error conditions, then call dlsym(), and then call dlerror() again, saving its return value into a variable, and check whether this saved value is not NULL.
```
https://linux.die.net/man/3/dlsym

Reply:
done in 35452f5